monkey/monkey_island/cc/server_utils/encryption/key_based_encryptor.py

```diff
@@ -29,19 +30,16 @@ def __init__(self, key: bytes):
     def encrypt(self, plaintext: str) -> str:
         cipher_iv = Random.new().read(AES.block_size)
         cipher = AES.new(self._key, AES.MODE_CBC, cipher_iv)
-        return base64.b64encode(cipher_iv + cipher.encrypt(self._pad(plaintext).encode())).decode()
+        return base64.b64encode(cipher_iv + cipher.encrypt(self._pad(plaintext))).decode()
 
     def decrypt(self, ciphertext: str):
         enc_message = base64.b64decode(ciphertext)
         cipher_iv = enc_message[0 : AES.block_size]
         cipher = AES.new(self._key, AES.MODE_CBC, cipher_iv)
-        return self._unpad(cipher.decrypt(enc_message[AES.block_size :]).decode())
+        return self._unpad(cipher.decrypt(enc_message[AES.block_size :]))
 
-    # TODO: Review and evaluate the security of the padding function
-    def _pad(self, message):
-        return message + (self._BLOCK_SIZE - (len(message) % self._BLOCK_SIZE)) * chr(
-            self._BLOCK_SIZE - (len(message) % self._BLOCK_SIZE)
-        )
+    def _pad(self, message: str) -> bytes:
+        return Padding.pad(message.encode(), self._BLOCK_SIZE)
 
-    def _unpad(self, message: str):
-        return message[0 : -ord(message[len(message) - 1])]
+    def _unpad(self, message: bytes) -> str:
+        return Padding.unpad(message, self._BLOCK_SIZE).decode()
```

Comment:
Since these functions are just thin wrappers, I suggest getting rid of them and inlining the calls to `Padding`.

```python
padded_plaintext = Padding.pad(...)
return base64.base64encode(cipher_iv + cipher.encrypt(padded_plaintext)).decode()

...

padded_plaintext = cipher.decrypt(...)
return Padding.unpad(padded_plaintext, self._block_size).decode()